cmd/ttn-lw-cli/commands/use.go

```diff
@@ -25,23 +25,40 @@ import (
 	"io/ioutil"
 	"os"
 	"path/filepath"
+	"regexp"
 
 	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
 	"go.thethings.network/lorawan-stack/v3/pkg/errors"
+	"go.thethings.network/lorawan-stack/v3/pkg/log"
 	"go.thethings.network/lorawan-stack/v3/pkg/rpcmiddleware/discover"
 	"gopkg.in/yaml.v2"
 )
 
 const configFileName = ".ttn-lw-cli.yml"
 
+func useCommandWriteConfigFlags() *pflag.FlagSet {
+	flagSet := &pflag.FlagSet{}
+	flagSet.Bool("user", false, "Write config file in user config directory")
+	flagSet.Bool("overwrite", false, "Overwrite existing config files")
+	return flagSet
+}
+
 var (
 	errNoHost     = errors.DefineInvalidArgument("no_host", "no host set")
 	errFileExists = errors.DefineAlreadyExists("file_exists", "`{file}` exists")
 	errFailWrite  = errors.DefinePermissionDenied("fail_write", "failed to write `{file}`")
-	useCommand    = &cobra.Command{
-		Use:               "use [host]",
+
+	validHostnameRegex             = regexp.MustCompile(`^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$`)
+	communityClusterRegex          = regexp.MustCompile(`^.cloud.thethings.network$`)
+	cloudClusterRegex              = regexp.MustCompile(`^([a-z][a-z0-9-]{2,}).([a-z][a-z0-9-]{2,}).cloud.thethings.industries$`)
+	cloudClusterMissingTenantRegex = regexp.MustCompile(`^([a-z][a-z0-9-]{2,}).cloud.thethings.industries$`)
```

Comment:
Or make the tenant ID in the previous optional and check if that group exists or not

Reply:
This is missing region right?